chainer/links/connection/lstm.py

```diff
@@ -70,6 +79,11 @@ def __call__(self, c, h, x):
                 output of LSTM units.
 
         """
+        if self.upward.has_uninitialized_params:
+            in_size = x.size // len(x.data)
```

Comment:
As `upward._initialize_params` does need `in_size`, we need `in_size` anyway. But as you pointed out, `self._initialize_params` does not use this argument. So, we should remove it from this method.